src/Symfony/Component/Security/Core/Authorization/Voter/VoterInterface.php

```diff
@@ -30,6 +30,9 @@ interface VoterInterface
      * This method must return one of the following constants:
      * ACCESS_GRANTED, ACCESS_DENIED, or ACCESS_ABSTAIN.
      *
+     * If more than one attribute is passed, the voter must grant access if access is granted to any of
```

Comment:
IMO this is not a hard requirement for any voter and it's totally fine to implement a custom voter that behaves differently. I would suggest that we move this comment into the docblock of the `Voter` class instead.

Reply:
I would be very careful with this approach, as it might bring a lot of confusion.

There clearly are conflicting views here. Here's how the Twig doc for `is_granted()` evolved in the last few months:

- on 9 Oct 2018, symfony/symfony-docs#10461 by @Jean85 changed the wording to:

    > Returns `true` if the current user has the required role, if only one is passed; if more than one is passed, with an array, the behavior depends on how the Access Decision Manager is configured; by default, the user has to have at least one of the passed roles.

- changed [the same day](https://github.com/symfony/symfony-docs/commit/c5d5dbb0a7f4229673af37f9705fa21353e501e8#diff-430dc9675fe0251899458375e09cb2db) by  @javiereguiluz to:

    > Returns `true` if the current user has the given role. If several roles are
passed in an array, returns ``true`` if the user has all of them or at least one
of them, depending on the value of this option:
`security.access_decision_manager.strategy`.

- on 25 Oct 2018, following [this comment on #10461](https://github.com/symfony/symfony-docs/pull/10461#pullrequestreview-168421115) by @xabbuh:

    > The access decision strategy defines how the result looks like when multiple voters voted on a certain attribute. It does not affect the internal behaviour of the RoleVoter.

    symfony/symfony-docs#10580 by @xabbuh changed it to:

    > Returns `true` if the current user has the given role. If several roles are passed in an array, `true` is returned if the user has at least one of them.

- on 18 July 2019, symfony/symfony-docs#11988 by @javiereguiluz changed it back to:

    > Returns `true` if the current user has the given role. If several roles are passed in an array, true is returned if the user has at least one of them or all of them, depending on the Access Decision Manager strategy.


As you can see, on the assumption that Twig's `is_granted()` follows the same rules as `VoterInterface::vote()`, there is no current consensus on how `is_granted()` should behave, and worse, there are regressions in the docs, ignoring relevant past comments:

- @Jean85 and @javiereguiluz think that the behaviour is influenced by the Access Decision Manager strategy, although @xabbuh [proved that it's not](https://github.com/symfony/symfony-docs/pull/10461#discussion_r228273161); @Jean85 [admitted that this was a mistake](https://github.com/symfony/symfony-docs/pull/10461#discussion_r228430204).
- regardless of the point above, @xabbuh now thinks that it's fine for voters to behave as they wish on this point, upvoted by @ro0NL 
- the [current Twig doc](https://github.com/symfony/symfony-docs/blob/4de3192439da0eeab74e270a1513b91eb2d08ee6/reference/twig_reference.rst) is clearly in infringement  with the source code anyway, as it says that:
    > `true` is returned if the user has at least one of them or all of them, depending on the Access Decision Manager strategy

    which directly contradicts [@xabbuh's argument](https://github.com/symfony/symfony-docs/pull/10461#discussion_r228273161):
    > the access decision strategy defines how the result looks like when multiple voters voted on a certain attribute (...) not the internal behaviour of the RoleVoter.
- and the [Voter](https://github.com/symfony/security-core/blob/4.3/Authorization/Voter/Voter.php#L41) class is opinionated about what should happen:
    > grant access as soon as at least one attribute returns a positive response

IMHO it was an unfortunate design decision to allow multiple values in the first place (forcing `isGranted('X') || isGranted('Y')` or `isGranted('X') && isGranted('Y')` would have probably been much less confusing), now one will always wonder what the behaviour is should they pass multiple values, and going to the docs or the source code will only make things more confusing, until they're all in sync on this point.

Now, to make things right without breaking BC, I would  recommend to apply the "less bad" fix:

- choose **one**, and only one, appropriate, *documented* behaviour here (most likely the *at least one of them* strategy)
- update **all** docs (Symfony Security component, Twig, source code) to reflect this fact
- maybe find a way to ensure that another PR a few months down the road won't again change this again (how?)